staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -305,40 +305,13 @@ var (
 		},
 	)
 	SuggestedFlowSchemaSystemLeaderElection = newFlowSchema(
-		"system-leader-election", "leader-election", 100,
+		"leader-election", "leader-election", 100,
 		flowcontrol.FlowDistinguisherMethodByUserType,
 		flowcontrol.PolicyRulesWithSubjects{
 			Subjects: append(
 				users(user.KubeControllerManager, user.KubeScheduler),
 				kubeSystemServiceAccount(flowcontrol.NameAll)...),
 			ResourceRules: []flowcontrol.ResourcePolicyRule{
-				resourceRule(
-					[]string{"get", "create", "update"},
-					[]string{corev1.GroupName},
-					[]string{"endpoints", "configmaps"},
-					[]string{"kube-system"},
-					false),
-				resourceRule(
-					[]string{"get", "create", "update"},
-					[]string{coordinationv1.GroupName},
-					[]string{"leases"},
-					[]string{flowcontrol.NamespaceEvery},
-					false),
-			},
-		},
-	)
-	SuggestedFlowSchemaWorkloadLeaderElection = newFlowSchema(
-		"workload-leader-election", "leader-election", 200,
-		flowcontrol.FlowDistinguisherMethodByUserType,
-		flowcontrol.PolicyRulesWithSubjects{
-			Subjects: kubeSystemServiceAccount(flowcontrol.NameAll),
-			ResourceRules: []flowcontrol.ResourcePolicyRule{
-				resourceRule(
-					[]string{"get", "create", "update"},
-					[]string{corev1.GroupName},
-					[]string{"endpoints", "configmaps"},
-					[]string{flowcontrol.NamespaceEvery},
```

Comment:
so workload that are still using `configmap` for leader election will need to use a different priority level if they want to avoid falling into `workload-low` priority level?

Reply:
Yes - this is another motivation to migrate your leader election to leases finally :)

Reply:
given the bad implications of delaying a read/write request to a leader election resource, this seems like a pretty big hammer to use to drive that change

Reply:
Well - that was more of a joke - that certainly isn't the motivation.

The motivation is that currently, all operations on endpoints and configmaps done by controller-manager (in particular endpoint-controller) is going against leader-election PL, which can starve actual leader election. And we've seen that.
The we're doing that to address actual bug.

Reply:
https://github.com/kubernetes/kubernetes/pull/106852 indicates that we'll offer two releases of transition time (very nice of us).  But during that transition, components will be acquiring two leases.  Having one of those backing lease objects be more prone to p&f failures than the other looks unfortunate to me.  What are the negative consequences of delaying this change for three releases (two for the supported migration, plus one more for handling skewed components)

Reply:
so currently, all changes to the Endpoints object (i.e. endpoints-controller) are going into leader-election bucket, which is very expensive
and especially in the mode of kube-proxy using Endpoints still, this was completely blowing up leader-election bucket if we increased the cost of those calls
all our core components were already migrated to use leases IIRC 3 releases ago, so this rule is only needed if:
- someone is explicitly overwriting the option to the legacy endpoint/configmap
0 has their own components
I think in both cases it's fair to say that the should add their own flow-schema [i.e. they still have a mitigation if they need]

Reply:
> and especially in the mode of kube-proxy using Endpoints still

Kube-proxy no longer uses endpoint, it uses endpoint slices only

Reply:
but the endpoints controller still writes to endpoints

Reply:
> Kube-proxy no longer uses endpoint, it uses endpoint slices only

Unless you have a cluster with (unsupported, but still) skew between control-plane and nodes...

> but the endpoints controller still writes to endpoints

Yeah - and always will for compatibility.
And putting all changes of endpoints/configmaps to "leader-election" PL just sounds like a bad choice.

[And we're not blocking anyone from adding new FlowSchema to catch that - you can always do that. It's only about the default.]

Reply:
> [And we're not blocking anyone from adding new FlowSchema to catch that - you can always do that. It's only about the default.]

Yes, but the defaults do matter.

> And putting all changes of endpoints/configmaps to "leader-election" PL just sounds like a bad choice.

You could segregate the updates from the endpoints controller while keeping this rule intact for three releases by using the endpoints controller identity, right?

I agree with the goal of coercing people to move to leases, but I'd prefer to do that using your change with leader election itself instead via inconsistent leader election loss under apiserver stress.

Reply:
> You could segregate the updates from the endpoints controller while keeping this rule intact for three releases by using the endpoints controller identity, right?

I guess so - forgive me my security ignorance, but each controller has its own service account, right?
[So that I can define proper subject matching only endpoint-controller with high precedentce: 
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/flowcontrol/v1beta2/types.go#L225 ]

That said, endpoint-controller is the main thing, but I've seen people installing their own thing in kube-system namespace:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go#L334
which would cause similar problems...


> I agree with the goal of coercing people to move to leases, but I'd prefer to do that using your change with leader election itself instead via inconsistent leader election loss under apiserver stress.

As I mentioned, this was a bit of a joke - the goal of this change is certainly not to move people to use Lease (the other PR, and the next removing the endpointsleases and configmapsleases in 3 releases) should do the job in less invasive way.
The motivation for this PR is to better protect "leader-election". I think we've all see significant outages cause by the fact that some components weren't able to renew their leader lock. This PR is closing this hole of people being able to do that relatively easily.

Reply:
> each controller has its own service account, right?

Sadly, that is still an option --- and it appears to default to `false`.  See https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/ .

Reply:
if we're trying to target a policy to the endpoints controller to keep it from consuming the leader-election endpoints flowschema, selecting both the well-known kube-controller-manager user and the endpoints controller service account seems sufficient

Reply:
It's not my desired solution, but let me maybe update this PR to make some progress

Reply:
Resolving with the PR already updated to reflect this.